Sources/Plasma/FeatureLib/pfMetalPipeline/ShaderSrc/FixedPipelineShaders.metal

```diff
@@ -0,0 +1,703 @@
+    sampler sampler3  [[ sampler(2), function_constant(hasLayer3)  ]];
+    sampler sampler4  [[ sampler(3), function_constant(hasLayer4)  ]];
+    sampler sampler5  [[ sampler(4), function_constant(hasLayer5)  ]];
+    sampler sampler6  [[ sampler(5), function_constant(hasLayer6)  ]];
+    sampler sampler7  [[ sampler(6), function_constant(hasLayer7)  ]];
+    sampler sampler8  [[ sampler(7), function_constant(hasLayer8)  ]];
+};
+
+typedef struct
+{
+    float4 position [[position]];
+    float3 texCoord1 [[function_constant(hasLayer1)]];
+    float3 texCoord2 [[function_constant(hasLayer2)]];
+    float3 texCoord3 [[function_constant(hasLayer3)]];
+    float3 texCoord4 [[function_constant(hasLayer4)]];
+    float3 texCoord5 [[function_constant(hasLayer5)]];
+    float3 texCoord6 [[function_constant(hasLayer6)]];
+    float3 texCoord7 [[function_constant(hasLayer7)]];
+    float3 texCoord8 [[function_constant(hasLayer8)]];
+    //float4 normal;
+    half4 vtxColor [[ centroid_perspective ]];
+    half4 fogColor;
+    //float4 vCamNormal;
+} ColorInOut;
+    
+    
+typedef struct
+{
+    float4 position [[position, invariant]];
+    float3 texCoord1;
+} ShadowCasterInOut;
+
+vertex ColorInOut pipelineVertexShader(Vertex in [[stage_in]],
+                                       constant VertexUniforms & uniforms [[ buffer(    VertexShaderArgumentFixedFunctionUniforms) ]],
+                                       constant plMetalLights & lights [[ buffer(VertexShaderArgumentLights) ]],
+                                       constant float4x4 & blendMatrix1 [[ buffer(VertexShaderArgumentBlendMatrix1), function_constant(temp_hasOnlyWeight1) ]])
+{
+    ColorInOut out;
+    //we should have been able to swizzle, but it didn't work in Xcode beta? Try again later.
+    const half4 inColor = half4(in.color.b, in.color.g, in.color.r, in.color.a) / half4(255.0f);
+    
+    const half3 MAmbient = mix(inColor.rgb, uniforms.ambientCol, uniforms.ambientSrc);
+    const half4 MDiffuse = mix(inColor, uniforms.diffuseCol, uniforms.diffuseSrc);
+    const half3 MEmissive = mix(inColor.rgb, uniforms.emissiveCol, uniforms.emissiveSrc);
+    //const half4 MSpecular = half4(mix(inColor, uniforms.specularCol, uniforms.specularSrc));
+
+    half3 LAmbient = half3(0.0, 0.0, 0.0);
+    half3 LDiffuse = half3(0.0, 0.0, 0.0);
+
+    const float3 Ndirection = normalize(float4(in.normal, 0.0) * uniforms.localToWorldMatrix).xyz;
+    
+    float4 position = (float4(in.position, 1.0) * uniforms.localToWorldMatrix);
+    if(temp_hasOnlyWeight1) {
+        const float4 position2 = blendMatrix1 * float4(in.position, 1.0);
+        position = (in.weight1 * position) + ((1.0f - in.weight1) * position2);
+    }
+
+    for (size_t i = 0; i < lights.count; i++) {
+        constant const plMetalShaderLightSource *lightSource = &lights.lampSources[i];
+        if(lightSource->scale == 0.0h)
+            continue;
+        
+        //w is attenation
```

Comment:
Should this be "attenuation" instead of "attenation"? Also may be just me, but it might help to be clear that you're talking about `direction.w` in this comment, otherwise it's not obvious to me what `w` means in your comment.